SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,806 @@
+}
+
+- (void)sdl_handleTransportEventUpdate {
+    if ([self.stateMachine isCurrentState:SDLSecondaryTransportStateStarted]) {
+        // The system sent Transport Event Update frame prior to Start Service ACK! Just keep the information and do nothing here.
+        SDLLogD(@"Received TCP transport information prior to Start Service ACK");
+        return;
+    }
+    if (self.transportSelection != SDLTransportSelectionTCP) {
+        return;
+    }
+
+    if ([self.stateMachine isCurrentState:SDLSecondaryTransportStateConfigured] && [self sdl_isTCPReady]) {
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConnecting];
+    } else if ([self sdl_isTransportOpened]) {
+        // Disconnect current transport. If the IP address is available then we will reconnect immediately.
+        SDLLogD(@"TCP transport information updated, disconnecting current transport");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+    } else if ([self.stateMachine isCurrentState:SDLSecondaryTransportStateReconnecting]) {
+        SDLLogD(@"TCP transport information updated, aborting reconnection timer");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+    }
+}
+
+- (void)sdl_handleTransportRegistered {
+    // secondary transport is now ready
+    [self.stateMachine transitionToState:SDLSecondaryTransportStateRegistered];
+}
+
+- (void)sdl_handleTransportRegisterFailed {
+    [self.stateMachine transitionToState:SDLSecondaryTransportStateReconnecting];
+}
+
+- (void)sdl_handleTransportRegisterTimeout {
+    // if the state is still Connecting, go back to Configured state and retry immediately
+    if ([self.stateMachine isCurrentState:SDLSecondaryTransportStateConnecting]) {
+        SDLLogD(@"Retry secondary transport connection after registration timeout");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+    }
+}
+
+- (void)sdl_handleProtocolClosed {
+    if ([self sdl_isTransportOpened]) {
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateReconnecting];
+    }
+}
+
+- (void)sdl_handleAppBecomeActive {
+    if (([self.stateMachine isCurrentState:SDLSecondaryTransportStateConfigured])
+        && self.transportSelection == SDLTransportSelectionTCP && [self sdl_isTCPReady]) {
+        SDLLogD(@"Resuming TCP transport since the app becomes foreground");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConnecting];
+    }
+}
+
+- (void)sdl_handleAppResignedActive {
+    if ([self sdl_isTransportOpened] && self.transportSelection == SDLTransportSelectionTCP) {
+        SDLLogD(@"Disconnecting TCP transport since the app will go to background");
+        [self.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+    }
+}
+
+- (BOOL)sdl_isTransportOpened {
+    return [self.stateMachine isCurrentState:SDLSecondaryTransportStateConnecting] ||
```

Comment:
Move `||` down to the next line

Reply:
Fixed in commit c6d6d98423e3372a89ad1f52331bd5f1b8c65fd6. Let me know if you want to align `||` after `return`.